src/Features/CSharp/Portable/IntroduceVariable/CSharpIntroduceParameterCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp.CodeGeneration;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.IntroduceVariable;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Simplification;
+
+namespace Microsoft.CodeAnalysis.CSharp.IntroduceVariable
+{
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp), Shared]
+    internal partial class CSharpIntroduceParameterCodeRefactoringProvider : AbstractIntroduceParameterService<
+        ExpressionSyntax,
+        InvocationExpressionSyntax,
+        IdentifierNameSyntax>
+    {
+        [ImportingConstructor]
+        [SuppressMessage("RoslynDiagnosticsReliability", "RS0033:Importing constructor should be [Obsolete]", Justification = "Used in test code: https://github.com/dotnet/roslyn/issues/42814")]
+        public CSharpIntroduceParameterCodeRefactoringProvider()
+        {
+        }
+
+        protected override SyntaxNode GenerateExpressionFromOptionalParameter(IParameterSymbol parameterSymbol)
+        {
+            return ExpressionGenerator.GenerateExpression(parameterSymbol.Type, parameterSymbol.ExplicitDefaultValue, canUseFieldReference: true);
```

Comment:
do you have any tests that exercise `canUseFieldReference: true`?

Reply:
No and I think I misinterpreted it's meaning. It should be false right since each identifier needs to map to either a parameter or its a constant?

Reply:
no, it's ok to reference a field then (fields can be constant).  wehre this would be used is if you had something like `void Foo(int x = int.MaxValue)`.  In this case, if you needed to inject a expr for that constant value, it would emit `int.MaxValue` instead of `2,147,483,647`

Reply:
(reminder to add test on this).

Reply:
(reminder to add test on this).

Reply:
sorry I forgot because I had not loaded all the comments, will add soon

Reply:
added